internal/cachehash/hash.go

```diff
@@ -66,7 +73,18 @@ func JSONFile(path string) (string, error) {
 	if err := json.Compact(buf, b); err != nil {
 		return "", redact.Errorf("compact json for hashing: %v", err)
 	}
-	return Bytes(buf.Bytes())
+	return Bytes(buf.Bytes()), nil
 }
 
 func newHash() hash.Hash { return sha256.New() }
+
+// Slug returns a deterministic URL slug version of the string. With the
+// following characteristics:
+//
+// * A 6 character hash is appended to avoid collisions
+// * Trims the slug to last 50 characters
+// * Removes any prefixed hashes to ensure first character is alphanumeric
+func Slug(s string) string {
```

Comment:
Where is this called? It sounds like this might be distinct from cache key hashing, no?

Reply:
It's no longer needed (now called internally by cache). Will remove.

In a follow up PR we can also start referencing cachehash and redact from `pkg` repo.